sunkit_instruments/rhessi/rhessi.py

```diff
@@ -350,3 +351,53 @@ def _build_energy_bands(label, bands):
     unit = matched.group('UNIT').strip()
 
     return [f'{band} {unit}' for band in bands]
+
+
+def hsi_fits2map(image_datacube):
+    """
+    Extracts single map images from a RHESSI flare image datacube.
+    Currently assumes input to be 4D.
+
+    Parameters
+    ----------
+    image_datacube: `str`
+        Path or URL to image datacube .fits
+
+    Returns
+    -------
+    `dict` of `list` of `sunpy.map.Map`
+        Each energy band has a list of maps where the index of the lists represent the time step
+    """
+    # import sunpy.map in here so that net and timeseries don't end up importing map
+    from sunpy.map import Map
+    
+    f = sunpy.io.read_file(image_datacube)
```

Comment:
Just following on from @nabobalis  comment above about checking for 4d cube - you could check the `NAXIS` keyword in the f[0]. I would also add in a check to make sure its a RHESSI imagecube, e.g. checking that the `INSTRUME` keyword is `RHESSI`

Reply:
Oh I see... But then the 4D assumption seems a bit restrictive to be honest.

Reply:
But this is just a function for RHESSI imagecubes, which will always be in this format, so I think it should be ok. 

A single RHESSI fits file can be passed directly to sunpy.map.Map

Reply:
Cool, good to know 👍